src/main/java/org/isf/menu/gui/UserBrowsing.java

```diff
@@ -279,17 +264,43 @@ public void ancestorAdded(AncestorEvent event) {
 				try {
 					if (answer == JOptionPane.YES_OPTION) {
 						userBrowsingManager.deleteUser(selectedUser);
-						userList.remove(table.getSelectedRow());
 						model.fireTableDataChanged();
 						table.updateUI();
 					}
+				} catch (OHDataIntegrityViolationException ex) {
+					answer = MessageDialog.yesNo(null, "angal.userbrowser.softdeleteuser.fmt.msg", selectedUser.getUserName());
+					if (answer == JOptionPane.YES_OPTION) {
+						try {
+							selectedUser.setDeleted(true);
+							userBrowsingManager.updateUser(selectedUser);
+							model.fireTableDataChanged();
+							table.updateUI();
+						} catch (OHServiceException e) {
+							OHServiceExceptionUtil.showMessages(e);
+						}
+					}
 				} catch (OHServiceException e) {
 					OHServiceExceptionUtil.showMessages(e);
 				}
 			}
 		});
 		buttonPanel.add(buttonDelete);
 
+		table.getSelectionModel().addListSelectionListener(
+			new ListSelectionListener() {
+
+				public void valueChanged(ListSelectionEvent e) {
+					if (table.getSelectedRow() >= 0) {
+						User selected = (User) table.getValueAt(table.getSelectedRow(), -1);
+						buttonEdit.setEnabled(!selected.isDeleted());
```

Comment:
I don't think we should disable this button, because one may need to restore the user, hence we need to update the UserEdit window with the new checkbox